### ISSUE TITLE
MusicVAE - Latent vector Z question

### ISSUE THREAD

**Issue by Gnefi** [tensorflow/magenta#1249]
Hi everyone!

I read you're paper and I studied you're code.
Only one little question: what is z? I mean I tried to "play" with you're code in my machine.
I trained the model on my own dataset and I found this z , a "matrix" 8x256 where 8 is the number of samples generated from trained model. 
My question is: what is every columns of this matrix? Music attribute of that sample? I don't understand this part.
Thank you and congratulations for you're incredible work. It's amazing :D

P.S: I'm a beginner in machine learning. Sorry :P 
Please don't judge me.

**Comment by adarob** [comment]
Hi, thanks for the question!

The dimensions of Z aren't actually anything specific. The latent space is a learned feature space and its dimensions are not constrained to have any specific meaning. However, the variational loss forces the space to be structured, which makes it possible to determine directions in this space that align to meaningful attributes. This is what the section of the paper on attribute vectors is discussing. 

I'd recommend you look at https://distill.pub/2017/aia if you want to learn a bit more about latent spaces.

Feel free to reopen if you have more questions.
